lib/SPIRV/SPIRVLowerBool.cpp

```diff
@@ -110,6 +110,17 @@ class SPIRVLowerBool : public ModulePass, public InstVisitor<SPIRVLowerBool> {
       I.setOperand(0, Sel);
     }
   }
+  virtual void visitSIToFPInst(SIToFPInst &I) {
+    auto Op = I.getOperand(0);
+    if (isBoolType(Op->getType())) {
+      auto Ty = Type::getInt32Ty(*Context);
+      auto Zero = getScalarOrVectorConstantInt(Ty, 0, false);
+      auto One = getScalarOrVectorConstantInt(Ty, 1, false);
+      assert(Zero && One && "Couldn't create constant int");
+      auto Sel = SelectInst::Create(Op, One, Zero, "", &I);
+      I.setOperand(0, Sel);
```

Comment:
This code is copy-pasted in this file several times, I suggest that we outline it into some helper function and re-use it in other `visit*` methods